server/src/main/java/org/apache/druid/catalog/model/PropertyAttributes.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.catalog.model;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.catalog.model.ModelProperties.PropertyDefn;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Definition and access of attributes of table definition properties. These
+ * are meta-attributes: attributes of attributes. These are primarily used to
+ * indicate the roll of each table property when used in a SQL table function.
+ */
+public class PropertyAttributes
+{
+  public static final String IS_SQL_FN_PARAM_KEY = "sqlFnArg";
+  public static final String IS_SQL_FN_OPTIONAL = "optional";
```

Comment:
Where are non-optional parameters validated to exist? I assume that any non-optional parameters that are not set should cause a user failure with respective message telling what parameter was missing?